### PR TITLE
Makefile: always build linux binaries when building for an image

### PR DESCRIPTION
**Pull request by estroz**
**Description of the change:** always build binaries destined for images with a linux target

**Motivation for the change:** no other OS targets can run in images

/cc @camilamacedo86 @joelanford 

**Checklist**

If the pull request includes user-facing changes, extra documentation is required:
- [ ] Add a new changelog fragment in `changelog/fragments` (see [`changelog/fragments/00-template.yaml`](https://github.com/operator-framework/operator-sdk/tree/master/changelog/fragments/00-template.yaml))
- [ ] Add or update relevant sections of the docs website in [`website/content/en/docs`](https://github.com/operator-framework/operator-sdk/tree/master/website/content/en/docs)

### CODE REVIEW

**Comment by estroz** [comment]
No changelog needed since builds using the new Makefile have never been used for a release.

/kind bug

File: Makefile [hunk 1]
```diff
@@ -76,10 +76,14 @@ image-build: $(foreach i,$(IMAGE_TARGET_LIST),image/$(i)) ## Build all images.
...
-	docker build -t $(IMAGE_REPO)/$*:dev -f ./images/$*/Dockerfile build/_image
-	@rm -rf build/_image
+image/%: BUILD_DIR = build/_image
+# Images run on the linux kernel, so binaries must always target linux.
+image/%: export GOOS = linux
+image/%: image-clean build/%
+	docker build -t $(IMAGE_REPO)/$*:dev -f ./images/$*/Dockerfile $(BUILD_DIR)
+
+image-clean:
+	rm -rf $(BUILD_DIR)
```
Comment (joelanford) [inline Makefile:86]: Running `image-clean` directly is the same as `rm -rf build`, right? If so, that's a bit confusing.

Can we run `rm -rf $(BUILD_DIR)` after the docker call?

```suggestion
	rm -rf $(BUILD_DIR)
```

**Review by joelanford — APPROVED** [review]
/lgtm

But I'll defer to @camilamacedo86's approval so she can test on her Mac.

**Review by jmrodri — APPROVED** [review]
This makes sense, I guess this has largely worked because most of us are using a Linux machien to build them.

**Pull request merged**